File: bindings/java/src/main/java/tech/turso/core/TursoEncryptionCipher.java
package tech.turso.core;

/**
 * Supported encryption ciphers for local database encryption.
 */
public enum TursoEncryptionCipher {
    /** AES-128-GCM cipher */
    AES_128_GCM("aes128gcm"),
    /** AES-256-GCM cipher */
    AES_256_GCM("aes256gcm"),
    /** AEGIS-256 cipher */
    AEGIS_256("aegis256"),
    /** AEGIS-256X2 cipher */
    AEGIS_256X2("aegis256x2"),
    /** AEGIS-128L cipher */
    AEGIS_128L("aegis128l"),
    /** AEGIS-128X2 cipher */
    AEGIS_128X2("aegis128x2"),
    /** AEGIS-128X4 cipher */
    AEGIS_128X4("aegis128x4");

    private final String value;

    TursoEncryptionCipher(String value) {
        this.value = value;
    }

    /**
     * Returns the string value of the cipher for internal use.
     */
    public String getValue() {
        return value;
    }
}

File: bindings/dotnet/src/Turso.Raw/Public/Value/TursoEncryptionCipher.cs
namespace Turso.Raw.Public.Value;

/// <summary>
/// Supported encryption ciphers for local database encryption.
/// </summary>
public enum TursoEncryptionCipher
{
    /// <summary>AES-128-GCM cipher</summary>
    Aes128Gcm,
    /// <summary>AES-256-GCM cipher</summary>
    Aes256Gcm,
    /// <summary>AEGIS-256 cipher</summary>
    Aegis256,
    /// <summary>AEGIS-256X2 cipher</summary>
    Aegis256x2,
    /// <summary>AEGIS-128L cipher</summary>
    Aegis128l,
    /// <summary>AEGIS-128X2 cipher</summary>
    Aegis128x2,
    /// <summary>AEGIS-128X4 cipher</summary>
    Aegis128x4,
}

internal static class TursoEncryptionCipherExtensions
{
    public static string ToRustString(this TursoEncryptionCipher cipher)
    {
        return cipher switch
        {
            TursoEncryptionCipher.Aes128Gcm => "aes128gcm",
            TursoEncryptionCipher.Aes256Gcm => "aes256gcm",
            TursoEncryptionCipher.Aegis256 => "aegis256",
            TursoEncryptionCipher.Aegis256x2 => "aegis256x2",
            TursoEncryptionCipher.Aegis128l => "aegis128l",
            TursoEncryptionCipher.Aegis128x2 => "aegis128x2",
            TursoEncryptionCipher.Aegis128x4 => "aegis128x4",
            _ => throw new ArgumentOutOfRangeException(nameof(cipher), cipher, null)
        };
    }
}
